overlord/overlord.go

```diff
@@ -102,6 +103,7 @@ func New() (*Overlord, error) {
 	o.hookMgr = hookMgr
 	o.stateEng.AddManager(o.hookMgr)
 
+	hooks.SetupHooks(o.hookMgr)
```

Comment:
I see, I think I misunderstood your suggestion. 
I moved hooks.go into hookstate package so hooks for snapstate are now registered from hookstate.Manager.